components/gitpod-protocol/src/util/gitpod-host-url.ts

```diff
@@ -24,21 +23,11 @@ const workspaceUrlPrefixRegex = RegExp(`^(([0-9]{4,6}|debug)-)?${baseWorkspaceID
 export class GitpodHostUrl {
     readonly url: URL;
 
-    constructor(urlParam?: string | URL) {
-        if (urlParam === undefined || typeof urlParam === "string") {
-            this.url = new URL(urlParam || "https://gitpod.io");
-            this.url.search = "";
-            this.url.hash = "";
-            this.url.pathname = "";
-        } else if (urlParam instanceof URL) {
-            this.url = urlParam;
-        } else {
-            log.error("Unexpected urlParam", { urlParam });
-        }
-    }
-
-    static fromWorkspaceUrl(url: string) {
```

Comment:
cc @geropl because of https://gitpod.slack.com/archives/C045J13P8A1/p1682420417312409?thread_ts=1682337644.081519&cid=C045J13P8A1